tests/integration/test_s3.py

```diff
@@ -333,6 +333,15 @@ def test_unicode_key_put_list(self):
             Bucket=self.bucket_name, Key=key_name)
         self.assertEqual(parsed['Body'].read().decode('utf-8'), 'foo')
 
+    def test_unicode_system_character(self):
+        # Verify we can use a unicode system character which would normally
+        # break the xml parser
+        key_name = 'foo\x08'
+        self.create_object(key_name)
+        parsed = self.client.list_objects(Bucket=self.bucket_name)
+        self.assertEqual(len(parsed['Contents']), 1)
+        self.assertEqual(parsed['Contents'][0]['Key'], key_name)
```

Comment:
Could we also do a second call where we specify ``EncodingType=url`` and show that it is not being url decoded? You could just add it to this test.